core-web/libs/portlets/dot-ema/src/lib/edit-ema-editor/components/edit-ema-palette/edit-ema-palette.component.ts

```diff
@@ -0,0 +1,127 @@
+import { Subject } from 'rxjs';
+
+import { AsyncPipe, NgIf } from '@angular/common';
+import {
+    ChangeDetectionStrategy,
+    Component,
+    EventEmitter,
+    Input,
+    OnDestroy,
+    OnInit,
+    Output,
+    inject
+} from '@angular/core';
+
+import { DotESContentService } from '@dotcms/data-access';
+import { DotPageContainerStructure } from '@dotcms/dotcms-models';
+
+import { EditEmaPaletteContentTypeComponent } from './components/edit-ema-palette-content-type/edit-ema-palette-content-type.component';
+import { EditEmaPaletteContentletsComponent } from './components/edit-ema-palette-contentlets/edit-ema-palette-contentlets.component';
+import { DotPaletteStore, PALETTE_TYPES } from './store/edit-ema-palette.store';
+
+@Component({
+    selector: 'dot-edit-ema-palette',
+    standalone: true,
+    templateUrl: './edit-ema-palette.component.html',
+    styleUrls: ['./edit-ema-palette.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    imports: [
+        NgIf,
+        AsyncPipe,
+        EditEmaPaletteContentTypeComponent,
+        EditEmaPaletteContentletsComponent
+    ],
+    providers: [DotESContentService, DotPaletteStore]
+})
+export class EditEmaPaletteComponent implements OnInit, OnDestroy {
+    @Input() languageId: number;
+    @Input() containers: DotPageContainerStructure;
+
+    @Output() dragStart = new EventEmitter();
+    @Output() dragEnd = new EventEmitter();
+
+    private readonly store = inject(DotPaletteStore);
+    private destroy$ = new Subject<void>();
+
+    vm$ = this.store.vm$;
+
+    PALETTETYPE = PALETTE_TYPES;
+
+    ngOnInit() {
+        this.store.loadAllowedContentTypes({ containers: this.containers });
+    }
+
+    /**
+     * Event handler for the drag start event.
+     * @param event The drag event.
+     */
+    onDragStart(event: DragEvent) {
+        this.dragStart.emit(event);
+    }
+
+    /**
+     * Handles the drag end event.
+     * @param event The drag event.
+     */
+    onDragEnd(event: DragEvent) {
+        this.dragEnd.emit(event);
+    }
+
+    /**
+     * Shows contentlets from a specific content type.
+     *
+     * @param contentTypeName - The name of the content type.
+     */
+    showContentletsFromContentType(contentTypeName: string) {
+        this.store.loadContentlets({
+            filter: '',
+            languageId: this.languageId.toString(),
+            contenttypeName: contentTypeName
+        });
+
+        this.store.changeView(PALETTE_TYPES.CONTENTLET);
+    }
+
+    /**
+     * Shows the content types in the palette.
+     */
+    showContentTypes() {
+        this.store.changeView(PALETTE_TYPES.CONTENTTYPE);
+        this.store.resetContentlets();
```

Comment:
Same here, but I'm not sure if it's correct to add the logic to reset the contentlets inside the `changeView`